Add CopySelectionTooltip component

kamicut opened the pull request:
PR to close #359 

<img width="1083" height="799" alt="image" src="https://github.com/user-attachments/assets/a9e6eb55-ef4e-41b9-bb93-ddbd44ef282f" />

@LanesGood @faustoperez the designs show a yellow AI sparkle, is that `secondary.500`?

kamicut committed:
Add CopySelectionTooltip component

kamicut committed:
fix icon fill and color

01painadam commented:
Noice! LGTM with a few small additions:

- On ctrl/cmd+c also dismiss the component
- I think the component should be positioned in the centre of the pane and not expand/shrink as the text selected (since thsi causes an overflow outside the pane)
- I think the sparkle needs a new colour in theme: `lime.500.value = "#C3D16F"` and should be solid fill

LanesGood commented on app/components/CopySelectionTooltip.tsx:
```diff
@@ -0,0 +1,141 @@
+          ref={tooltipRef}
+        >
+          <Flex align="center" gap="2">
+            <Icon as={SparkleIcon} color="fg.muted" />
```
```suggestion
            <Icon as={SparkleIcon} color="secondary.500" />
```
@kamicut yes, secondary.500

01painadam reviewed (approved):
LGTM, made a few suggestions above but not blocking!

kamicut committed:
Fixes

- Center tooltip on repaint
- Enable copy text on mobile
- Cmd+c to dismiss tooltip

kamicut commented:
I've added:
- cmd+c to dismiss
- center tooltip
- I've enabled "copy text" on mobile, however I'm not too happy with this because it negatively interacts with the draggable drawer

01painadam commented:
Nice, let's merge! @kamicut could you add a follow up ticket in the project to solve the mobile issue - I don't think we need to solve now but good to keep track
